lib/src/script.dart

```diff
@@ -153,6 +153,28 @@ class Script {
   /// script exits.
   final _outputCloser = StreamCloser<List<int>>();
 
+  /// The script's signal handler to terminate the process.
+  bool Function(ProcessSignal)? _signalHandler;
+
+  /// Sends a [ProcessSignal] to terminate the process.
+  ///
+  /// If the [Script] is a Unix-style OS process, pass the given signal to the
+  /// process. On platforms without signal support, including Windows, the
+  /// [signal] is ignored and the process terminates in a platform specific way.
+  ///
+  /// Returns `true` if the signal is successfully delivered to the process.
```

Comment:
```suggestion
  /// Returns `true` if the signal is successfully delivered to the script.
```

Not all scripts are processes